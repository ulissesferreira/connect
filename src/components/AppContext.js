import React from 'react';

export default React.createContext({
  loggedIn: false,
  user: {},
  login: () => {},
  logout: () => {},
  saveUserInfo: () => {},
  setSearchQuery: () => {},
  setProfilePic: () => {},
  showToast: () => {},
  searchQuery: '',

  resetSearchQuery: false,
})