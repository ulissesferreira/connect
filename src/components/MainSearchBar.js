import React, { Component } from 'react';
import { Link } from 'react-router-dom';

import * as Api from '../utils/Api';

export default class MainSearchBar extends Component {

  constructor(props) {
    super(props)
    this.state = {
      search: '',
      expertise: [],
      company: [],
      people: []
    }
  }

  handleChange = (event) => {
    this.setState({
      search: event.target.value
    }, () => {
      if (this.state.search && this.state.search.length > 1) {
        // if (this.state.search.length % 2 === 0) {
        this.getInfo()
        // }
      } else if (!this.state.query && this.state.search.length === 1) {
        this.setState({
          expertise: [],
          company: [],
          people: []
        })
      }
    })
  }

  getInfo = () => {
    Api.searchQuick(this.state.search).then((res) => {
      console.log(res)
      this.setState({
        expertise: res.expertise,
        company: res.company,
        people: res.people
      })
    })
  }

  expertise = (props) => {
    if (!props) {
      return 
    }
    console.log(props.results)
    const expertise = props.map((element) => (
      <Link to={'/expertise/' + element.name}>
        <li key={element.name}>
          {element.name}
          {element.description}
        </li>
      </Link>
    ))
    return <ul>{expertise}</ul>
  }

  people = (props) => {
    if (!props) {
      return
    }
    console.log(props.results)
    const people = props.map((element) => (
      <Link to={'/people/' + element.keycode}>
        <li key={element.name}>
          {element.name}
          {element.keycode}
          {element.bio}
          {element.profilePic}
        </li>
      </Link>
    ))
    return <ul>{people}</ul>
  }

  company = (props) => {
    if (!props) {
      return
    }
    console.log(props.results)
    const company = props.map((element) => (
      <Link to={'/company/' + element.name}>
        <li key={element.name}>
          {element.name}
          {element.description}
        </li>
      </Link>
    ))
    return <ul>{company}</ul>
  }

  render() {
    return (
      <div>
        <input onChange={this.handleChange} />
        {this.expertise(this.state.expertise)}
        {this.people(this.state.people)}
        {this.company(this.state.company)}
      </div>
    );
  }
}