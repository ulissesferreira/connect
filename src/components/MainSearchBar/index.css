#search-input {
  border: 1px solid var(--light-grey);
  border-radius: .25rem;
  outline: none;
}