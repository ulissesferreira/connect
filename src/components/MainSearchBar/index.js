import React, { Component } from 'react';
import Api from '../../utils/Api';
import './index.css';

export default class MainSearchBar extends Component {

  constructor(props) {
    super(props)
    this.state = {
      search: props.search ? props.search : '',
    }
  }

  handleChange = (event) => {
    this.setSearch(event.target.value)
  }

  setSearch = (search) => {
    this.setState({
      search: search
    }, () => {
      if (this.state.search === '') {
        Api.getAllMentors().then((res) => {
          this.props.setMentors(res.mentors)
        })
      } else {
        Api.searchFull(this.state.search).then((res) => {
          this.props.setMentors(res.search)
        })
      }
    })

    // update query in parent component
    this.props.searchChanged(search)
  }

  render() {
    return (
      <input type='text' id="search-input" className='icon' placeholder="Try looking for a person..."
        onChange={this.handleChange} />
    );
  }
}