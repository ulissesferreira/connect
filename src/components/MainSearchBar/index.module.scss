.SearchWrapper{
    width: 80%;
}

//remove important after removing the input section from the big css file
.input {
    background-color: #f1f3f4;
    height: 3rem; 
    outline: none !important;
    border: 1px solid transparent !important; 
    border-radius: 100px !important;
    padding-left: 50px !important; 
    width: 50vw;
    background-image: url('data:image/svg+xml;utf8,<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" fill="%235f6368" viewBox="0 0 24 24"><path d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"/><path d="M0 0h24v24H0z" fill="none"/></svg>');
    background-repeat: no-repeat;
    background-position: 2.25% 50%;
    transition: .25s ease all;
    min-width: 200px;
    max-width: 600px;
    
  }
  
  .input::placeholder {
    color: #5f6368!important;
  } 
  
