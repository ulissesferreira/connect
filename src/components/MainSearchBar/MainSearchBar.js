import React, { Component } from 'react'
import { Redirect } from 'react-router-dom'

import AppContext from '../AppContext'

import styles from './MainSearchBar.module.scss'

export default class MainSearchBar extends Component {
  static contextType = AppContext
  constructor(props) {
    super(props)
    this.state = {
      searchQuery: '',
    }
  }

  handleChange = event => {
    this.setState({
      searchQuery: event.target.value,
    })
  }

  handleKeyPress = event => {
    if (event.key === 'Enter') {
      this.context.setSearchQuery(this.state.searchQuery)
      this.context.startSearchQuery(true)
    }
  }

  RedirectToMain = () => {
    if (this.context.isSearchQuery) {
      return <Redirect to="/" />
    }
  }

  render() {
    return (
      <div className={styles.SearchWrapper}>
        <input
          type="text"
          className={styles.input}
          placeholder="What are you looking for?"
          onChange={this.handleChange}
          value={this.state.searchQuery}
          onKeyDown={this.handleKeyPress}
        />
        {this.RedirectToMain()}
      </div>
    )
  }
}
