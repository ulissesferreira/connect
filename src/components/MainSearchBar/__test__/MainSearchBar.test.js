import React from 'react'
import { render, fireEvent, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom'

import { BrowserRouter as Router } from 'react-router-dom'
import MainSearchBar from '../MainSearchBar'
import AppContext from '../../AppContext'

afterEach(cleanup)

const renderContext = (mockSearchQuery, mockSetSearchQuery) => {
  return render(
    <Router>
      <AppContext.Provider
        value={{
          startSearchQuery: mockSearchQuery,
          setSearchQuery: mockSetSearchQuery,
        }}
      >
        <MainSearchBar />
      </AppContext.Provider>
    </Router>
  )
}

describe('component is rendered correctly', () => {
  it('rendered the input', () => {
    const { getByPlaceholderText } = render(
      <Router>
        <MainSearchBar />
      </Router>
    )
    const input = getByPlaceholderText('What are you looking for?')
    expect(input).toBeTruthy()
  })

  it('take a snapshot', () => {
    const { container } = render(
      <Router>
        <MainSearchBar />
      </Router>
    )
    expect(container).toMatchSnapshot()
  })
})

describe('check search query context', () => {
  it('check if search query context', () => {
    const mockSearchQuery = jest.fn(searchQuery => true)
    const mockSetSearchQuery = jest.fn()
    const { getByPlaceholderText } = renderContext(
      mockSearchQuery,
      mockSetSearchQuery
    )
    const input = getByPlaceholderText('What are you looking for?')
    fireEvent.change(input, { target: { value: 'test' } })
    fireEvent.keyDown(input, { key: 'Enter', code: 13 })
    expect(mockSetSearchQuery.mock.calls[0][0]).toBe('test')
  })
})
