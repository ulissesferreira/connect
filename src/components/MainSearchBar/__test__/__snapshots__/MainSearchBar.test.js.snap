// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component is rendered correctly renders the component 1`] = `
<div>
  <div
    class="SearchWrapper"
  >
    <input
      class="input"
      placeholder="What are you looking for?"
      type="text"
      value=""
    />
  </div>
</div>
`;
